src/npc/npc_update.cpp

```diff
@@ -5159,6 +5178,8 @@ void UpdateNPCs()
                                         }
                                     }
                                 }
+
+                                // @Wohlstand, should we add something here for the BGO fence fix?
```

Comment:
BGO fence fix doesn't require any structure side fixes, it needs the fix at the player logic and at layer processing logic, which was done already.

Reply:
The problem is that this could allow the `SpeedX` of the BGOs to remain non-zero forever, since there is nowhere else to set the `SpeedX` of the BGOs.

Reply:
But here, this is a case where the NPC is setting the layer's speed, so I'm not sure if we need to do anything here. Let's talk about this separately.